WordPress/Classes/Services/MediaSettings.swift

```diff
@@ -1,13 +1,95 @@
 import Foundation
+import AVFoundation
 
 class MediaSettings: NSObject {
     // MARK: - Constants
     fileprivate let maxImageSizeKey = "SavedMaxImageSizeSetting"
     fileprivate let removeLocationKey = "SavedRemoveLocationSetting"
+    fileprivate let maxVideoSizeKey = "SavedMaxVideoSizeSetting"
+
 
     fileprivate let minImageDimension = 150
     fileprivate let maxImageDimension = 3000
 
+    enum VideoResolution: String {
+        case size640x480 = "AVAssetExportPreset640x480"
+        case size960x540 = "AVAssetExportPreset960x540"
+        case size1280x720 = "AVAssetExportPreset1280x720"
+        case size1920x1080 = "AVAssetExportPreset1920x1080"
+        case size3840x2160 = "AVAssetExportPreset3840x2160"
+        case sizeOriginal = "AVAssetExportPresetPassthrough"
+
+        var videoPreset: String {
+            switch (self) {
+            case(.size640x480):
+                return AVAssetExportPreset640x480
+            case(.size960x540):
+                return AVAssetExportPreset960x540
+            case(.size1280x720):
+                return AVAssetExportPreset1280x720
+            case(.size1920x1080):
+                return AVAssetExportPreset1920x1080
+            case(.size3840x2160):
+                return AVAssetExportPreset3840x2160
+            case(.sizeOriginal):
+                return AVAssetExportPresetPassthrough
+            }
+        }
+
+        var description: String {
+            switch (self) {
+            case(.size640x480):
+                return NSLocalizedString("VGA (640x480)", comment: "Indicates a video will be resized to 640x480 when uploaded.")
+            case(.size960x540):
+                return NSLocalizedString("Quarter HD (960x540)", comment: "Indicates a video will be resized to 960x540 when uploaded.")
+            case(.size1280x720):
+                return NSLocalizedString("HD (1280x720)", comment: "Indicates a video will be resized to HD 1280x720 when uploaded.")
+            case(.size1920x1080):
+                return NSLocalizedString("Full HD (1920x1080)", comment: "Indicates a video will be resized to Full HD 1920x1080 when uploaded.")
+            case(.size3840x2160):
+                return NSLocalizedString("4K(3840x2160)", comment: "Indicates a video will be resized to 4K 3840x2160 when uploaded.")
+            case(.sizeOriginal):
+                return NSLocalizedString("Original", comment: "Indicates a video will use its original size when uploaded.")
+            }
+        }
+
+        var intValue: Int {
+            switch (self) {
```

Comment:
I don't really like the repetition of these switch cases switching on or returning literal numbers of 1, 2, 3, 4, 5, 6 etc. But I can't think of many better ways of doing it. Perhaps to make it clearer, we could just move these into two methods on `VideoSizingRow`: `sliderValueForResolution`, and `resolutionForSliderValue` or something?